### DISCUSSION TITLE
Disable Binary Warning for Single Lines?

### DISCUSSION THREAD

**Discussion by LangLangBart** [sharkdp/bat#2971] [Q&A]
```sh
bat --version
bat 0.24.0
```

Is there a way to disable the `binary` warning message? For example, I have a line of text with a `NUL` byte, and piping it into `bat` reveals a warning. Interestingly, placing the `NUL` byte on a second line won't show the warning.

```sh
❌
printf "First\0" | bat -p
[bat warning]: Binary content from STDIN will not be printed to 
the terminal (but will be present if the output of 'bat' is piped). 
You can use 'bat -A' to show the binary file contents.
```

```sh
✅
printf "First\nSecond\0" | bat -p
  First
  Second^@
```

---

Using the `--show-all/-A` flag doesn't show the warning but is not useful for my use case.

I try to colorize my `zsh` history and pipe it into `fzf`.

If the last command was a single line, then `bat` fails; if the last command was multiline, then `bat` succeeds as intended.

```sh
# zsh only, the '-N'  flag separates the array elements by `NUL`
print -rN -- ${(uv)history[@]} | bat -pl sh | fzf --read0
```

---

**EDIT1:** Related #823

**Comment by keith-hall** [comment]
as you've already found out, `bat` will show that warning when the content is detected as binary and the header style is not being displayed and not using the show-all flag: https://github.com/sharkdp/bat/blob/8f8c953ab69403a7acf516e65265b7f70bc3ec73/src/printer.rs#L435-L444

there is no other way to remove this warning right now, and indeed you don't just want to disable the warning but to output the actual first line. Perhaps pre-processing the input would be a suitable solution - fake it all as being multiline by adding a newline to the end of each entry being passed into `bat`...

**Reply by LangLangBart** [reply]
Thanks for pointing out the source code. The decision to label the input as `BINARY` seems to be made in `src/input.rs`:

https://github.com/sharkdp/bat/blob/8f8c953ab69403a7acf516e65265b7f70bc3ec73/src/input.rs#L260-L271


> fake it all

Making the first line empty and in a later step remove that line seems to work. 

```sh
# zsh
(print; print -rNC1 -- "${(@uv)history}") | bat -pl zsh | sed '1d' | fzf --read0
```

It's a bit _hacky_. Do you think it's worth opening an issue requesting a new flag, e.g. `--read0`, for `bat` so that the `content_type` is set to `None` when the flag is present?

**Reply by keith-hall** [reply]
sure, go ahead and make an issue for it, and we will hopefully get some input from others :)

**Reply by LangLangBart** [reply]
Thanks for the input. I will close the discussion.

https://github.com/sharkdp/bat/issues/2974
